pkg/tracing/tracing.go

```diff
@@ -0,0 +1,154 @@
+package tracing
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	"go.opentelemetry.io/otel"
+	"go.opentelemetry.io/otel/exporters/jaeger"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace"
+	"go.opentelemetry.io/otel/exporters/otlp/otlptrace/otlptracegrpc"
+	"go.opentelemetry.io/otel/exporters/stdout/stdouttrace"
+	"go.opentelemetry.io/otel/propagation"
+	"go.opentelemetry.io/otel/sdk/resource"
+	"go.opentelemetry.io/otel/sdk/trace"
+	semconv "go.opentelemetry.io/otel/semconv/v1.20.0"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type tracer struct {
+	Provider *trace.TracerProvider
+	Shutdown func()
+}
+
+func Setup(svc, ttype string) (func(), error) {
+	ctx := context.Background()
+
+	var (
+		tracer *tracer
+		err    error
+	)
+
+	tracer, err = NewTracerProvider(ctx, svc, ttype)
+	if err != nil {
+		return nil, err
+	}
+
+	otel.SetTracerProvider(tracer.Provider)
+	otel.SetTextMapPropagator(
+		propagation.NewCompositeTextMapPropagator(
+			propagation.TraceContext{},
+			propagation.Baggage{},
+		),
+	)
+
+	return func() { tracer.Shutdown() }, nil
+}
+
+func NewTracerProvider(ctx context.Context, svc, ttype string) (*tracer, error) {
+	switch ttype {
+	case "otlp":
```

Comment:
should ttype be a const or enum?

Reply:
maybe enum?

Reply:
actually, I'll go with const for now since there are no much difference